labellines/line_label.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import matplotlib.patheffects as patheffects
+import numpy as np
+from matplotlib.text import Text
+
+if TYPE_CHECKING:
+    from datetime import datetime
+    from typing import Any, Union
+
+    from matplotlib.axes import Axes
+    from matplotlib.lines import Line2D
+
+    Position = Union[float, datetime, np.datetime64]
```

Comment:
For the sake of consistency
```suggestion
    Position = float | datetime | np.datetime64
```